libsolidity/ast/Types.cpp

```diff
@@ -669,7 +669,23 @@ tuple<bool, rational> RationalNumberType::isValidLiteral(Literal const& _literal
 			// parse the exponent
 			bigint exp = bigint(string(expPoint + 1, _literal.value().end()));
 
-			if (exp > numeric_limits<int32_t>::max() || exp < numeric_limits<int32_t>::min())
+			// SOL-008: disallow gigantic exponents to limit memory consumption
+			bigint bytesNeeded = string(_literal.value().begin(), expPoint).length() + abs(exp);
```

Comment:
Please don't create a new string just to count the length. Also, why does this formula result in the number of bytes required for representing the value?

Reply:
I think it is perfectly fine to parse both the mantissa and the exponent. Perhaps it would be easier to put this check right before the call to `boost::multiprecision::pow`?